src/fetch/dune.py

```diff
@@ -59,27 +59,45 @@ async def fetch(self, query: Query) -> list[DuneRecord]:
             log.error(f"Got {err} - Exiting")
             sys.exit()
 
-    async def latest_app_hash_block(self) -> int:
+    async def latest_app_hash_block(self, chain: str) -> int:
         """
         Block Range is used to app hash fetcher where to find the new records.
         block_from: read from file `fname` as a loaded singleton.
             - uses genesis block is no file exists (should only ever happen once)
             - raises RuntimeError if column specified does not exist.
         block_to: fetched from Dune as the last indexed block for "GPv2Settlement_call_settle"
         """
+        if chain == "mainnet":
+            return int(
+                # KeyError here means the query has been modified and column no longer exists
+                # IndexError means no results were returned from query (which is unlikely).
+                (await self.fetch(QUERIES["LATEST_APP_HASH_BLOCK"].query))[0][
+                    "latest_block"
+                ]
+            )
         return int(
             # KeyError here means the query has been modified and column no longer exists
             # IndexError means no results were returned from query (which is unlikely).
-            (await self.fetch(QUERIES["LATEST_APP_HASH_BLOCK"].query))[0][
+            (await self.fetch(QUERIES["LATEST_APP_HASH_BLOCK_GNOSIS"].query))[0][
                 "latest_block"
             ]
         )
```

Comment:
using an `if .. elif ...` seems clearer here, throwing an exception is the chain name is not correct.
(the pythonic way might be to use a `match ... case ... case ...` block)